cohttp-lwt-jsoo/src/cohttp_lwt_jsoo.ml

```diff
@@ -202,15 +202,18 @@ module Make_client_async (P : Params) = Make_api (struct
                         (fun s -> `String s)
                     in
                     if xhr_response_supported then
-                      Js.Opt.case
-                        (File.CoerceTo.arrayBuffer xml##.response)
-                        (fun () ->
-                          Firebug.console##log
-                            (Js.string
-                               "XHR Response is not an arrayBuffer; using \
-                                responseText");
-                          respText ())
-                        (fun ab -> `ArrayBuffer ab)
+                      if Js.Opt.return xml##.response == Js.null then
```

Comment:
- Does this need to be done also at: https://github.com/mirage/ocaml-cohttp/pull/761/files#diff-17a4a55959fc8c74371af346bb1da1f88718607b785b1f1ca651b3b226d3f3cfR311
- If so, can we extract this function instead of having so much duplication?
- Is there a way to test this?

Reply:
Sure, IIRC there's even a comment by @avsm there that it is duplicated.

Reply:
I gave it a try, the only difference is the async version wrapping the response in a `Lwt.t`, so we'd need to parameterize the whole thing over the "future" monad. But really the whole `Make_client_sync` and `Make_client_async` are 99% duplicated, a simple abstraction over that monad would factor it (someone has been veeeery lazy or in a hurry when writing this 😅).

I don't mind taking a look, but honestly that should be a different PR.

Reply:
Regarding the reproduction, I know of no easy way. AFAICT it's the internal XHR implementation of chrome, and it's not clear when to me why `responseType` is `"arraybuffer"`, but `response` is `null`. It looks like a mild bug [according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseType), but since it's meant to be used from javascript and everything fits everywhere™ in that magical world, it's probably not much of an issue - and by that I mean not a bigger issue that the rest of the language and libraries.

I can't reproduce it myself, I just get that exception reported a few times a day from our app running in the wild, all I can say is that it's possible to reach this state with a fairly recent chrome (electron 12). I thought I had a reproduction in the js test suite, which prompted me to drop this PR, but I was empirically wrong it seems.

Reply:
Fair enough. Given how complicated is managing exceptions across ocaml and js, I think it makes sense to be a bit proactive here. We should probably add a LOG message that the call returned a `null` response though, so it does not come unnoticed if it happens by mistake

Reply:
> I don't mind taking a look, but honestly that should be a different PR.

That's good for me. Thanks for putting in the effort!

Reply:
Done